src/System.Private.Interop/src/Shared/__ComObject.cs

```diff
@@ -126,7 +126,7 @@ public unsafe class __ComObject : ICastable
 
         // if m_additionalCachedInterfaces_dontAccessDirectly == CacheLocked, the cache is being updated and
         // cannot be read or written from another thread.  We do this instead of using a "real" lock, to save space.
-        private static readonly AdditionalComInterfaceCacheContext[] CacheLocked = new AdditionalComInterfaceCacheContext[0];
+        private static readonly AdditionalComInterfaceCacheContext[] CacheLocked = Array.Empty<AdditionalComInterfaceCacheContext>();
```

Comment:
This won't get ever any benefits from sharing - because of AdditionalComInterfaceCacheContext is private type and you can see all places where it is used.

Also, this code plays tricks with object instance identify - it is more robust (less bug prone) to actually create an instance in such code instead of getting shared one.